include --glob flag in rimraf

connorskees opened the pull request:
Running `lerna run clean --scope=@fluidframework/merge-tree` currently crashes with the error `Error: Illegal characters in path.`. Adding the `--glob` flag to `rimraf` fixes this crash. I've only verified this change works for the merge-tree package, but the change should be straightforward for the other packages.

connorskees committed:
include --glob flag in rimraf

Abe27342 commented:
looks good. only 2 concerns here:

1. Do we need a policy-check update? would rather not add 153 warnings :)
2. Does this make the experience work on unix/linux?

probably just need someone with a mac or linux dev machine (maybe also WSL for good measure) to confirm 2nd one.


...and while I was typing my comment:

3. what's going on with our link checker? (I'm assuming that's completely unrelated to this PR, but i'll investigate)

Abe27342 commented:
answer to (3) is that https://fluidframework.com has a link to the fluidframework twitter (which... I guess is a thing) in the footer, and the linkchecker appears to give an error for each page on fluidframework.com. so the output makes sense assuming twitter was down for a moment.

Abe27342 commented:
> I'll double check on my personal linux machine that the command removes the right files.
> 
> For the first point, what is a policy-check update?

We have a lot of 'handlers' implemented that ensure our package.json scripts stay relatively uniform (i.e. alphabetical, reasonable deps installed, have a readme, etc. that kind of thing). I wasn't sure if we had one that enforced the contents of the `clean` script were reasonable, but I checked just now and I don't see any reference that looks troublesome. There's e.g. some logic which invokes the clean script in `fluidBuild.ts`, and some which verifies build scripts are accompanied by clean scripts in `fluidPackageCheck.ts`, but that's about it.

(I should mention that both of those files are in fluid-build and not actually policy-check; if you wanted to look over the handlers we have implemented open the file `fluidCase.ts` and look around that folder)

Abe27342 reviewed (approved):
LGTM once you confirm this doesn't break the experience on unix

connorskees commented:
Tested on linux box, no crashes and deletes the files I would expect it to. Will merge once CI is green

tylerbutler reviewed (commented):
@curtisman FYI. Will this break fluid-build's script parsing?

connorskees committed:
Merge branch 'main' of https://github.com/microsoft/FluidFramework into feat/glob-clean

tylerbutler commented:
/azp run Build - Client packages

tylerbutler commented:
We've updated to the latest build-tools in client, so as long as pnpm fluid-build works as expected with this change, ship it. :shipit:

curtisman commented:
> @curtisman FYI. Will this break fluid-build's script parsing?

Nope

connorskees committed:
merge main

connorskees committed:
sort package.json files

connorskees committed:
remove flag from snapshots package

connorskees committed:
merge main

CraigMacomber commented on azure/package.json:
```diff
@@ -0,0 +1,120 @@
+{
```
why is this file added. Seems unrelated. Bad merge?

connorskees replied:
Yes, bad merge. Intending to delete this before merging into main.

CraigMacomber commented on azure/packages/azure-client/package.json:
```diff
@@ -22,7 +22,7 @@
 		"build:test": "tsc --project ./src/test/tsconfig.json",
 		"ci:build:docs": "api-extractor run",
-		"clean": "rimraf dist lib *.tsbuildinfo *.build.log",
+		"clean": "rimraf --glob dist lib *.tsbuildinfo *.build.log",
```
I believe before this change, this command worked on some OSes (like linux) because the shell was expanding the "*". It seems like if we are going to use --glob, for consistency it would be better to have rimraf expand it on all platforms, so we should quote the patterns to stop the shell from doing it.

connorskees committed:
merge main

connorskees committed:
add quotes around paths

connorskees committed:
remove file from bad merge

connorskees committed:
revert change to snapshot file

connorskees committed:
revert change inside docs

the build for this package appears to be broken from a pre-existing
issue. modifying the contents of the docs/ folder causes CI to run for
this package and fail

connorskees committed:
revert change in odsp driver

connorskees committed:
revert changes to snapshot package

connorskees committed:
un-revert some changes
